### PR TITLE
feat: implement drawLine on Mask with bresenham-zingl

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/image-js/image-js-typescript/issues/125?notification_referrer_id=NT_kwDOA2I9VLMzOTMyNDY0NjY4OjU2NzY5ODc2#issuecomment-1173807865

### CODE REVIEW

**Comment by wadjih-bencheikh18** [comment]
the test `roi/getConvexHullMask/random shape, filled = false` is now failing

**Comment by targos** [comment]
@wadjih-bencheikh18 Océane doesn't work this week. Can you look into why it fails? Maybe the new behavior is better than before.

**Comment by wadjih-bencheikh18** [comment]
> @wadjih-bencheikh18 Océane doesn't work this week. Can you look into why it fails? Maybe the new behavior is better than before.

yes ofc no problem

File: src/draw/drawLineOnIjs.ts [hunk 1]
```diff
@@ -40,13 +40,15 @@ export function drawLineOnIjs(
...
     bitDepth: [8, 16],
   });
+  const numberChannels = Math.min(newImage.channels, color.length);
 
   if (from.column === to.column && from.row === to.row) {
+    for (let channel = 0; channel < numberChannels; channel++) {
+      newImage.setValue(from.column, from.row, channel, color[channel]);
+    }
     return newImage;
   }
```
Comment (targos) [inline src/draw/drawLineOnIjs.ts:50]: I think you can just remove the `if` and let the library handle this case?

File: src/draw/drawLineOnMask.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ export function drawLineOnMask(
   const newMask = maskToOutputMask(mask, options, { clone: true });
 
   if (from.column === to.column && from.row === to.row) {
+    newMask.setBit(from.column, from.row, 1);
     return newMask;
   }
```
Comment (targos) [inline src/draw/drawLineOnMask.ts:34]: same

File: src/draw/drawLineOnIjs.ts (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/draw/drawLineOnIjs.ts:50]: yes okey no problem

**Comment by wadjih-bencheikh18** [comment]
> the test `roi/getConvexHullMask/random shape, filled = false` is now failing @opatiny

about this 
i found that there's an example that return diffrent line between the library algo and our algorithm

line (0,2) to (4,0)
```
[0, 0, 0, 0, 0],
[0, 0, 0, 0, 0],
[0, 0, 0, 0, 0],
```


result of our algo
```
[0, 0, 0, 0, 1],
[0, 0, 1, 1, 0],
[1, 1, 0, 0, 0],
```

result of the library
```
[0, 0, 0, 1, 1],
[0, 1, 1, 0, 0],
[1, 0, 0, 0, 0],
```

File: src/draw/drawLineOnIjs.ts (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/draw/drawLineOnIjs.ts]: yes it handle the case

**Comment by targos** [comment]
The new result looks good to me (I think both are technically correct). @lpatiny ?

**Pull request merged**